service/lib/agama/autoyast/files_reader.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2024] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+# :nodoc:
+module Agama
+  module AutoYaST
+    # Builds the Agama "files" section from an AutoYaST profile.
+    class FilesReader
+      # @param profile [ProfileHash] AutoYaST profile
+      def initialize(profile)
+        @profile = profile
+      end
+
+      # Returns a hash corresponding to Agama "files" section.
+      #
+      # If there is no files-related information, it returns an empty hash.
+      #
+      # @return [Hash] Agama "files" section
+      def read
+        return {} if profile["files"].nil?
+
+        # files section contains list of file nodes.
+        # supported file sub-elements:
+        # - file_location (a file source) (-> source)
+        # - file_path (this one seems to be the only mandatory,
+        #             a file destination or directory to be created if ends with '/')
+        #             (-> destination)
+        # - file_owner (-> owner)
+        # - file_permissions (permissions)
+        #
+        # unsupported for now:
+        # - file_content
+        # - file_script
+        files = profile.fetch_as_array("files")
+
+        files_json = files.reduce([]) do |res, file|
+          # Currently we support only url to file source
+          file["source"] = file.delete("file_location")
+          file["destination"] = file.delete("file_path")
+          file["permissions"] = file.delete("file_permissions")
+          file["owner"] = file.delete("file_owner")
```

Comment:
same here, how it will look like? rust part now uses pair which I am not sure how it is serialized, but I can write own serializer that will use something like "root:root" string for serialization.

Reply:
in documentation is "tux.users"  so I can either create a tuple, array, hash,  or keep it as a string ....